File: 08-2-2-Adapter-Solution/Ducks/RedheadDuck.cs
namespace Adapter.Ducks
{
    using System;
    using Adapter.Fly;
    using Adapter.Quack;

    public sealed class RedheadDuck : Duck
    {
        public RedheadDuck()
            : base(new DuckQuackStrategy(), new RealFlyStrategy())
        {
        }

        public override void Display() => Console.WriteLine("Red head, gray body.");
    }
}

File: 09-DesignPatterns-Java/Solution/Adapter/src/Ducks/RedheadDuck.java
package Ducks;

import Fly.RealFlyStrategy;
import Quack.DuckQuackStrategy;

public class RedheadDuck extends Duck {
    public RedheadDuck() {
        super(new DuckQuackStrategy(), new RealFlyStrategy());
    }

    @Override
    public void display() {
        System.out.println("Red head, gray body.");
    }
}
